src/devices/src/virtio/queue.rs

```diff
@@ -22,6 +23,26 @@ pub(super) const VIRTQ_DESC_F_WRITE: u16 = 0x2;
 // The Virtio Spec 1.0 defines the alignment of VirtIO descriptor is 16 bytes,
 // which fulfills the explicit constraint of GuestMemoryMmap::read_obj_from_addr().
 
+#[derive(Debug)]
+pub enum QueueError {
+    /// Attempted to write to an invalid address into the used ring.
+    GuestMemoryError(vm_memory::GuestMemoryError),
+}
+
+impl fmt::Display for QueueError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::QueueError::*;
+
+        match &*self {
+            GuestMemoryError(e) => write!(
+                f,
+                "Failed to write value into the virtio queue used ring: {}",
```

Comment:
I think this should be a generic guest mem error description so it'd be applicable to other cases not just the virtio queue used ring.

`GuestMemory access error: {}` sounds good to me for example

Reply:
Good point.

Reply:
I think that errors generated in specific code can benefit from being very explicit at the cost of increasing the binary size.

Reply:
We can keep the specific error description but then we should also change the error name to `QueueError::UsedRing` or smth specific.